configurator/data/run.sh

```diff
@@ -4,8 +4,34 @@ set -e
 DIRSFIRST=$(bashio::config 'dirsfirst')
 ENFORCE_BASEPATH=$(bashio::config 'enforce_basepath')
 IGNORE_PATTERN="$(bashio::jq "/data/options.json" ".ignore_pattern")"
+
+DEPLOYMENT_KEY=$(bashio::config 'deployment_key')
+DEPLOYMENT_KEY_PROTOCOL=$(bashio::config 'deployment_key_protocol')
+
 WAIT_PIDS=()
 
+# Adds the SSH deployment key specified in the config, for use by git
+function add-ssh-key {
+    echo "[Info] Start adding SSH key"
+    mkdir -p ~/.ssh
+
+    (
+        echo "Host *"
+        echo "    StrictHostKeyChecking no"
+    ) > ~/.ssh/config
+
+    echo "[Info] Setup deployment_key on id_${DEPLOYMENT_KEY_PROTOCOL}"
+    while read -r line; do
+        echo "$line" >> "${HOME}/.ssh/id_${DEPLOYMENT_KEY_PROTOCOL}"
+    done <<< "$DEPLOYMENT_KEY"
+
+    chmod 600 "${HOME}/.ssh/config"
+    chmod 600 "${HOME}/.ssh/id_${DEPLOYMENT_KEY_PROTOCOL}"
+}
+
+# Add SSH key for git, if defined
+if [[ ! -z ${DEPLOYMENT_KEY} ]]; then add-ssh-key; fi
```

Comment:
Use Bashio for this, e.g.:

`if bashio::config.has_value "deployment_key"; then add-ssh-key; fi`

Reply:
Done